src/prefect/blocks/core.py

```diff
@@ -1050,3 +1050,15 @@ def __new__(cls: Type[Self], **kwargs) -> Self:
             object.__setattr__(m, "__dict__", kwargs)
             object.__setattr__(m, "__fields_set__", set(kwargs.keys()))
             return m
+
+    def get_block_placeholder(self) -> Optional[str]:
+        """
+        Returns the block placeholder for the current block which can be used for
+        templating.
+
+        If a block has not been saved, the return value will be `None`.
```

Comment:
```suggestion
        If the block has not been saved, the return value will be `None`.
```

Reply:
I don't know how you intend to use this so this is just a general design question, but would it be better to `ValueError` if the block isn't saved to raise early?

Reply:
Raising an exception felt a little aggressive in this case, but it would be more explicit to raise an exception rather than the caller needing to implicitly handle the case where `None` is returned. I'll create a custom exception for the case where the block isn't saved because I think it'll make it easier to write better error messages in `deploy`.